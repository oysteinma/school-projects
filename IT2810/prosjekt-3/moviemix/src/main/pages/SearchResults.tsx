import { Box } from "@mui/material";
import CardView from "../components/CardView";
import MovieMixHeader from "../components/MovieMixHeader";
import Search from "../components/Search";

/**
 *
 * @returns the SearchResults page, which displays the search results
 */
export default function SearchResults() {
  return (
    <Box
      display={"flex"}
      flexDirection={"column"}
      alignItems={"center"}
      alignContent={"center"}
      width={"50vw"}
      mx={"auto"}
      marginTop={"5vh"}
      marginBottom={"5vh"}
      gap={"5vh"}
    >
      <MovieMixHeader title="Search Results" />
      <Search />
      <CardView />
    </Box>
  );
}
