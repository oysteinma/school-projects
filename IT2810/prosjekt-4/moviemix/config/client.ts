import { ApolloClient, NormalizedCacheObject } from "@apollo/client";

import { cache } from "./cache";

/**
 * Setup Apollo Client
 */
export const client: ApolloClient<NormalizedCacheObject> = new ApolloClient({
  cache: cache,
  uri: "https://it2810-59s.idi.ntnu.no/graphql",
});
