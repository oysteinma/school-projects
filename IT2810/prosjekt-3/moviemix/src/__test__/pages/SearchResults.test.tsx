import SearchResults from "../../main/pages/SearchResults";
import { render } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import { HashRouter as Router } from "react-router-dom";

/**
 * @tests The search results page component - namely that the component renders
 */
it("check if SearchResults renders succesfully", () => {
  const { container } = render(
    <MockedProvider>
      <Router>
        <SearchResults />
      </Router>
    </MockedProvider>
  );
  expect(container).toMatchSnapshot();
});
