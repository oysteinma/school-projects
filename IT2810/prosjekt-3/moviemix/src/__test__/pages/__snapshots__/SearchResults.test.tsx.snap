// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if SearchResults renders succesfully 1`] = `
<div>
  <div
    class="MuiBox-root css-ynm7vf"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 css-1nhhm1w-MuiTypography-root"
    >
      Search Results
    </h1>
    <div
      class="MuiBox-root css-bnbs76"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
              style="width: 40vw; height: 50px; background: white; border-radius: 0;"
            >
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </div>
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
                id="searchbar"
                placeholder="Search"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-kbyzs3-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Search
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
      style="width: 80vw;"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-10 css-vlkmuj-MuiGrid-root"
      />
    </div>
  </div>
</div>
`;
